### PR TITLE
Prepare for 2.0 release

### PR DESCRIPTION
**Pull request by orta**
This is the one.

Mid-April 2015, I wrote down in an issue https://github.com/artsy/mobile/issues/31#issue-70116307 that we really should start automating some of the cultural rules we were establishing in Artsy. 

6 Months later in September, I wrote the first commit https://github.com/danger/danger/commit/54bff6406a9f3a880f00e14c13a2314f9e7173e0. Within a hour or so I had ripped enough code from CocoaPods to make a working prototype https://github.com/danger/danger/commit/0a429b426f6d803ec1ecc7b9c662620bad559043. By the end of the day, it did all of the basics https://github.com/danger/danger/commit/f574143f96f7b3260e30915591d702612c511812. By start of 2016 it was in production at Arsty https://github.com/artsy/eigen/pull/1013.

@KrauseFx and I played "who owns Danger" for a while, and eventually managed to get an org for the project. My `origin` remote for Danger still points to `https://github.com/krausefx/danger/`.

Since then we've expanded to support:

* Running on  Circle, Travis, Jenkins, Buildkite, Semaphore, TeamCity, Xcode Bots, Drone, Surf.
* An amazing plugin system, bootstrapped by @KrauseFx  and @dbgrandi 
* Windows `;)`
* All sorts of work flows
* 25 contributors, over 1000 commits across the projects

and soon GitLab and Bitbucket.

My last month of work on Danger - started with some README driven design, https://github.com/danger/danger.systems/commit/a7f0cb2628b44c6b356ad031a6c7ca558e00eecd covering all of the concept for http://danger.systems. This has been one of the biggest automation project I've worked on, it's ready in case Danger [gets big](http://artsy.github.io/blog/2016/07/03/handling-big-projects/). Danger Systems is a really cool project that should really help ease the amount of support requests we get. More docs = 👯 .

I even have a Danger [spotify playlist](https://open.spotify.com/user/ortatherox/playlist/39vTSdxtIKzpEybyvTs3Ll).

---

Thanks everyone who has helped out, this would be such a different project without you all.

----

_Why 2.0?_ Well, it's a final homage to Danger. She's the real 1.0 - can't try to replicate her.

I want to wait till tomorrow to merge this, and announce Danger is ready. 👍

### CODE REVIEW

**Comment by orta** [comment]
what is up with these fails?

**Comment by KrauseFx** [comment]
```
lib/danger/commands/init_helpers/interviewer.rb:75:24: C: Omit parentheses for ternary conditions.
          the_answer = (i == 0) ? answer.underline : answer
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/danger/commands/init_helpers/interviewer.rb:75:25: C: Use i.zero? instead of i == 0.
          the_answer = (i == 0) ? answer.underline : answer
                        ^^^^^^
lib/danger/danger_core/dangerfile.rb:52:5: C: When using method_missing, define respond_to_missing?.
    def method_missing(method_sym, *arguments, &_block) ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/danger/danger_core/dangerfile.rb:198:17: C: Use (status[:errors] + status[:warnings] + status[:messages] + status[:markdowns]).count.zero? instead of (status[:errors] + status[:warnings] + status[:messages] + status[:markdowns]).count == 0.
      return if (status[:errors] + status[:warnings] + status[:messages] + status[:markdowns]).count == 0
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/danger/plugin_support/plugin.rb:20:5: C: When using method_missing, define respond_to_missing? and fall back on super.
    def method_missing(method_sym, *arguments, &block) ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/danger/request_source/github.rb:114:19: C: Use errors.count.zero? instead of errors.count == 0.
        status = (errors.count == 0 ? 'success' : 'failure')
                  ^^^^^^^^^^^^^^^^^
70 files inspected, 6 offenses detected
```

Interesting that only Travis failed the build.

**Comment by orta** [comment]
I think it's not using the bundle'd rubocop, those linters got added in the last few days. I've bumped our bundler version, and fixed them

File: lib/danger/plugin_support/plugin.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@ def self.instance_name
 
     # Since we have a reference to the Dangerfile containing all the information
     # We need to redirect the self calls to the Dangerfile
+
+    # rubocop:disable Style/MethodMissing
     def method_missing(method_sym, *arguments, &block)
```
Comment (segiddins) [inline lib/danger/plugin_support/plugin.rb]: you need to implement `respond_to_missing?` here

Comment (orta) [inline lib/danger/plugin_support/plugin.rb]: in the next release, I'm not adding something I don't understand entirely an hour or two before release.

**Comment by orta** [comment]
Alright, I'm calling this green, it works locally and it looks like `travis/pr` seems to have locked on two of our ruby versions.

screen shot 2016-07-25 at 11 21 00 am

**Pull request merged**